docs/source/tutorial/Introduction.lhs

```diff
@@ -0,0 +1,65 @@
+# Defining GraphQL type APIs
+
+First some imports:
+
+``` haskell
+{-# LANGUAGE DataKinds #-}
+{-# LANGUAGE TypeOperators #-}
+{-# LANGUAGE OverloadedStrings #-}
+module Introduction where
+
+import Protolude
+
+import GraphQL.TypedSchema (Object, Field, Argument, (:>))
+import GraphQL.TypedApi (HandlerType, (:<>)(..))
+```
+
+The core idea for this library is that we define a composite type that
+specifies the whole API, and then implement a matching handler.
+
+The main GraphQL entities we care about are Objects and Fields. Each
+Field can have arguments.
+
+``` haskell
+type HelloWorld = Object "HelloWorld" '[]
+  '[ Argument "greeting" Text :> Field "me" Text
+   ]
+```
```

Comment:
https://bitbucket.org/birkenfeld/pygments-main/issues/820/haskell-highlighting-promoted-types - bug from 2012. Will try to pick that up